forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -290,6 +307,36 @@ fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) {
     }
 }
 
+fn clean(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) {
+    {
+        let snapshot_dir = snapshot_dir
+            .clone()
+            .unwrap_or_else(|| default_snapshot_dir(config));
+        println!("Snapshot dir: {}", snapshot_dir.display());
+        let mut snapshots_to_delete = vec![];
+        if let Ok(dir) = fs::read_dir(snapshot_dir) {
+            dir.flatten()
+                .map(|entry| entry.path())
+                .filter(|p| p.extension().unwrap_or_default() == "car")
+                .for_each(|p| snapshots_to_delete.push(p));
+        }
```

Comment:
could we `collect` instead? Also, it'd be nice not to omit the error. To my understanding, in the case of a filesystem error (no rights to read for whatever reason, the directory doesn't exist) we would just get `no files to delete`.

It'd be nice to return a `Result` from these commands and provide a juicy error. I assume people using Forest are technical, so even a cryptic IO error should be acceptable.

Reply:
fixed